core/src/main/java/org/polypheny/db/type/entity/PolyInterval.java

```diff
@@ -27,32 +28,95 @@
 import org.jetbrains.annotations.Nullable;
 import org.polypheny.db.nodes.IntervalQualifier;
 import org.polypheny.db.type.PolySerializable;
+import org.polypheny.db.type.PolyType;
+import org.polypheny.db.type.entity.category.PolyNumber;
+import org.polypheny.db.util.temporal.TimeUnit;
 
 @EqualsAndHashCode(callSuper = true)
 @Value
 @Slf4j
+@NonFinal
 public class PolyInterval extends PolyValue {
 
 
-    public BigDecimal value;
-    public IntervalQualifier qualifier;
+    @NotNull
+    public Long millis;
+
+    @Getter
+    @NotNull
+    public Long months;
 
 
     /**
-     * Creates a PolyInterval.
+     * Creates a PolyInterval, which includes a millis and a month value, to allow for combinations liki 7-1 Month to Day, which is 7 months and 7 days.
+     * And required for example by the SQL standard.
```

Comment:
What is required by the SQL standard?

Reply:
https://github.com/polypheny/Polypheny-DB/pull/483#discussion_r1547196254 added in reformulation.